chore(prompt-experiments): set langfuse-native environment

marliessophie opened the pull request:
<!-- ELLIPSIS_HIDDEN -->


> [!IMPORTANT]
> Enhance event processing by introducing internal environment handling and refactor schema creation for Langfuse events.
> 
>   - **Behavior**:
>     - Introduces `isLangfuseInternal` option in `processEventBatch()` in `processEventBatch.ts` to handle internal Langfuse events.
>     - Updates `fetchLLMCompletion()` in `fetchLLMCompletion.ts` to use `isLangfuseInternal` for event processing.
>     - Adds `PROMPT_EXPERIMENT_ENVIRONMENT` constant in `types.ts` for internal environment setting.
>   - **Schemas**:
>     - Refactors schema creation in `types.ts` to support public and internal environments using `createIngestionEventSchema()`.
>     - Deprecates `ingestionEvent` in favor of `createIngestionEventSchema()`.
>   - **Tests**:
>     - Updates tests in `otelMapping.servertest.ts` and `IngestionService.integration.test.ts` to use `createIngestionEventSchema()`.
>     - Modifies `experimentsService.test.ts` to check for correct environment setting in traces.
>   - **Misc**:
>     - Bumps `langfuse-langchain` version to `3.38.3` in `package.json`.
>     - Removes duplicate export in `index.ts`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for 82948a43aa4c90053a61c1ece30c5fc8b24ea29a. You can [customize](https://app.ellipsis.dev/langfuse/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

marliessophie committed:
chore(prompt-experiments): set langfuse-native environment

marliessophie committed:
refactor: no longer export ingestionEvent schema directly

marliessophie commented:
WARNING: do not merge before making required changes in TS SDK and upgrading PR to new version.

marliessophie committed:
chore: rm line number

hassiebp commented on packages/shared/src/server/llm/fetchLLMCompletion.ts:
```diff
@@ -145,6 +145,9 @@ export async function fetchLLMCompletion(
         await processEventBatch(
```
I think now that we are adding three more positional args to processEventBatch, we should move to a single config object rather than 5 positional args that in turn need comments to show what they are for / reduce bug risk

marliessophie replied:
Agree, planned this too :)

marliessophie commented on packages/shared/src/server/ingestion/types.ts:
```diff
@@ -553,6 +553,34 @@ export const ingestionEvent = z.discriminatedUnion("type", [
+    return schema;
+  }
+
+  return schema.check((ctx) => {
+    // Override environment regex validation errors when isLangfuseInternal is true
+    const filteredIssues = ctx.issues.filter((issue) => {
+      // Remove issues related to environment field regex validation
+      const isEnvironmentRegexError =
+        issue.path?.includes("environment") &&
+        issue.message === ENVIRONMENT_NAME_REGEX_ERROR_MESSAGE;
+      return !isEnvironmentRegexError;
+    });
+
+    // Clear all issues and add back only the non-environment regex ones
+    ctx.issues.length = 0;
+    ctx.issues.push(...filteredIssues);
+  });
```
@maxdeichmann as discussed this feels a bit "unschön" - I will read docs and see if there is a cleaner zod-native solution that does not result in a performance penalty. If you have a better idea of the bat would love to hear it!

marliessophie replied:
Did not find a better approach from docs - would suggest timeboxing another 30 minutes for research and otherwise merge current solution, wdyt @maxdeichmann? :)

marliessophie committed:
chore(ingestion): refactor processEventBatch to accept options object

marliessophie committed:
chore(ingestion): refactor ingestion schemas for public and internal environments

- Introduced separate schemas for public and internal environments.
- Updated environment name validation logic and error messages.
- Refactored event schemas to utilize the new schema structure.
- Deprecated direct export of `ingestionEvent` schema in favor of factory method for better environment handling.

marliessophie committed:
chore: push

marliessophie committed:
Merge branch 'main' into marlies/lfe-4445-native-environment

marliessophie committed:
chore: type environment schema as string

marliessophie committed:
chore: push

marliessophie committed:
Merge branch 'main' into marlies/lfe-4445-native-environment

marliessophie committed:
chore: fix import statements

marliessophie commented on packages/shared/src/server/index.ts:
```diff
@@ -20,7 +20,6 @@ export * from "./clickhouse/schema";
 export * from "./repositories/definitions";
 export * from "../server/ingestion/types";
 export * from "./ingestion/processEventBatch";
-export * from "../server/ingestion/types";
```
Doublicated import statement

marliessophie committed:
chore: update langfuse-langchain to version 3.38.3

marliessophie committed:
Merge branch 'main' into marlies/lfe-4445-native-environment

marliessophie committed:
chore: prettier

marliessophie committed:
chore: revert prettier

marliessophie committed:
Merge branch 'main' into marlies/lfe-4445-native-environment
